web_widget_switchcase/static/src/js/web_widget_switchcase.js

```diff
@@ -0,0 +1,53 @@
+odoo.define('web_widget_switchcase', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var formats = require('web.formats');
+    var form_widgets = require('web.form_widgets');
+
+    var CharSwitchCase = form_widgets.FieldChar.extend({
+        transformations: ['upper', 'lower'],
+        init: function (field_manager, node) {
+            this._super(field_manager, node);
+            this.options = _.defaults(this.options, {
+                transform: this.transformations[0]
+            });
+            this.current_transformation = this.options['transform'];
+        },
+        initialize_content: function() {
+            this._super();
+            var self = this;
+            if(this.$input) {
+                this.$input.keyup(function(){
+                    var old_val = self.$input.val();
+                    var new_val = self.transform(old_val);
+                    self.$input.val(new_val);
+                });
+            }
+        },
+        parse_value: function (val, def) {
+            return this._super(this.transform(val), def);
+        },
+        format_value: function (val, def) {
+            return this._super(this.transform(val), def);
+        },
+        transform: function (val) {
+            if (!val)
+                return val;
+            switch (this.current_transformation) {
+                case 'upper':
+                    return val.toUpperCase();
+                case 'lower':
+                    return val.toLowerCase();
+                default:
+                    console.log(this.current_transformation + ' case unknown');

```

Comment:
Can you move this to the initialization of the widget (init function) and use console.error, please.
Also if the transformation is unknown return the non-transformed value to still make it work like a normal char widget.